srv/modules/runners/ready.py

```diff
@@ -71,17 +71,39 @@ def apparmor(self):
         """
         Scan minions for apparmor settings.  
         """
-        contents = self.local.cmd(self.search , 'cmd.shell', [ '/usr/sbin/apparmor_status --enabled 2>/dev/null; echo $?' ], expr_form="compound")
+        contents = self.local.cmd(self.search, 'cmd.shell',
+                                  [('/usr/sbin/aa-status --enabled '
+                                    '2>/dev/null; echo $?')],
+                                  tgt_type="compound")
+        minions_with_apparmor = 0
+        minions_with_profiles = 0
         for minion in contents:
             if contents[minion] and int(contents[minion]) == 0:
-                msg = "enabled on minion {}".format(minion)
-                if 'apparmor' in self.warnings:
-                    self.warnings['apparmor'].append(msg)
-                else:
-                    self.warnings['apparmor'] = [ msg ]
-        if 'apparmor' not in self.warnings:
+                minions_with_apparmor += 1
+                # check if all file.managed for the profiles would apply
+                # cleanly using test=True. If so salt returns 'result': True
+                # burried in its return structure
+                result = self.local.cmd(minion, 'state.apply',
+                                        ['ceph.apparmor.profiles'],
+                                        kwarg={'test': 'true'})
```

Comment:
I'm worried about calling minions sequentially.   Calling once in parallel we keep the runtime more consistent.  (We have this in #1104 too.)

Reply:
fair. Fixed in the PR to master...will update backport.